### PR TITLE
MAINTAINERS: add initial entries for west.yml projects

### PR DESCRIPTION
**Pull request by mbolivar-nordic**
This is part of the road towards replacing CODEOWNERS with
MAINTAINERS (tracked in the process working group as #38725).

As part of that work, we wanted to have a way to track the maintainers
of each west project (entry in west.yml) by name of project.

This is an initial attempt at that, based mostly on my personal
knowledge, the git logs, and some rough guesswork. I fully expect
omissions and errors, but it should be enough to get us started, with
fixes to follow incrementally from the people who know their
individual areas better than I do.

Signed-off-by: Martí Bolívar <marti.bolivar@nordicsemi.no>

### CODE REVIEW

**Review by laurenmurphyx64 — DISMISSED** [review]

**Review by stephanosio — COMMENTED** [review]
It bothers me a little that this file uses 4 spaces for indentation (which, I understand, is very questionable) yet the changes here use 2 spaces.

I guess we can update the whole file to use 2 spaces instead of 4 spaces later.

File: MAINTAINERS.yml [hunk 1]
```diff
@@ -1841,6 +1838,489 @@ West:
...
+
+"West project: tinycbor":
+  status: obsolete
+  files:
+    - modules/Kconfig.tinycbor
+  labels:
+    - manifest-tinycbor
+
+"West project: tinycrypt":
+  status: obsolete
```
Comment (carlescufi) [inline MAINTAINERS.yml:2261]: Hmm no, this is not obsolete. There are no viable plans or alternatives to replace it (yet).

File: MAINTAINERS.yml [hunk 2]
```diff
@@ -1841,6 +1838,489 @@ West:
...
+    - nashif
+  files:
+    - modules/Kconfig.cmsis
+    - modules/Kconfig.cmsis_dsp
+    - modules/Kconfig.cmsis_nn
+  labels:
+    - manifest-cmsis
+
+"West project: edtt":
+  status: odd fixes
```
Comment (carlescufi) [inline MAINTAINERS.yml:1873]: I wonder if @aescolar would volunteer to maintain this one?

File: MAINTAINERS.yml [hunk 3]
```diff
@@ -1841,6 +1838,489 @@ West:
...
+"West project: fatfs":
+  status: maintained
+  maintainers:
+    - de-nordic
+  files: []
+  labels:
+    - manifest-fatfs
+
+"West project: hal_altera":
+  status: obsolete
```
Comment (carlescufi) [inline MAINTAINERS.yml:1892]: Checking with @nashif about this one

File: MAINTAINERS.yml [hunk 4]
```diff
@@ -1841,6 +1838,489 @@ West:
...
+  status: maintained
+  maintainers:
+    - nandojve
+  files:
+    - modules/Kconfig.atmel
+  labels:
+    - manifest-hal_atmel
+
+"West project: hal_cypress":
+  status: odd fixes
```
Comment (carlescufi) [inline MAINTAINERS.yml:1908]: pinging @npal-cy, should you be the maintainer?

File: MAINTAINERS.yml [hunk 5]
```diff
@@ -1841,6 +1838,489 @@ West:
...
+    - gmarull
+  collaborators:
+    - soburi
+  files:
+    - modules/hal_gigadevice/
+  labels:
+    - manifest-hal_gigadevice
+
+"West project: hal_infineon":
+  status: odd fixes
```
Comment (carlescufi) [inline MAINTAINERS.yml:1940]: Same about this one @npal-cy

File: MAINTAINERS.yml [hunk 6]
```diff
@@ -1841,6 +1838,489 @@ West:
...
+  files:
+    - modules/Kconfig.telink
+  labels:
+    - manifest-hal_telink
+
+"West project: hal_ti":
+  status: odd fixes
+  collaborators:
+    - cfriedt
+    - gmarull
```
Comment (carlescufi) [inline MAINTAINERS.yml:2079]: @vaishnavachath would you like to be added here?

File: MAINTAINERS.yml [hunk 7]
```diff
@@ -1841,6 +1838,489 @@ West:
...
+  maintainers:
+    - Mani-Sadhasivam
+  files:
+    - modules/loramac-node/
+  labels:
+    - manifest-loramac-node
+
+"West project: lvgl":
+  status: odd fixes
+  files:
```
Comment (carlescufi) [inline MAINTAINERS.yml:2136]: @brgl would you like to maintain or be added as a collaborator?

File: MAINTAINERS.yml [hunk 8]
```diff
@@ -1841,6 +1838,489 @@ West:
...
+    - modules/mbedtls/
+  labels:
+    - manifest-mbedtls
+
+"West project: mcuboot":
+  status: maintained
+  maintainers:
+    - d3zd3z
+  collaborators:
+    - nvlsianpu
```
Comment (carlescufi) [inline MAINTAINERS.yml:2165]: Andrzej should be a maintainer

File: MAINTAINERS.yml [hunk 9]
```diff
@@ -1841,6 +1838,489 @@ West:
...
+    - rlubos
+  collaborators:
+    - pdgendt
+  files: []
+  labels:
+    - manifest-openthread
+
+"West project: segger":
+  status: odd fixes
+  files: []
```
Comment (carlescufi) [inline MAINTAINERS.yml:2230]: Can you add @nordic-krch as a collaborator here?

File: MAINTAINERS.yml [hunk 10]
```diff
@@ -1841,6 +1838,489 @@ West:
...
+    - modules/TraceRecorder/
+  labels:
+    - manifest-TraceRecorderSource
+
+"West project: trusted-firmware-m":
+  status: maintained
+  maintainers:
+    - microbuilder
+  collaborators:
+    - joerchan
```
Comment (carlescufi) [inline MAINTAINERS.yml:2281]: Please add @SebastianBoe as well

File: MAINTAINERS.yml [hunk 11]
```diff
@@ -1841,6 +1838,489 @@ West:
...
+    - modules/trusted-firmware-m/
+  labels:
+    - manifest-trusted-firmware-m
+
+"West project: tf-m-tests":
+  status: maintained
+  maintainers:
+    - microbuilder
+  collaborators:
+    - joerchan
```
Comment (carlescufi) [inline MAINTAINERS.yml:2292]: Please add @SebastianBoe as well

File: MAINTAINERS.yml [hunk 12]
```diff
@@ -1841,6 +1838,489 @@ West:
...
+  files: []
+  labels:
+    - manifest-tf-m-tests
+
+"West project: psa-arch-tests":
+  status: maintained
+  maintainers:
+    - microbuilder
+  collaborators:
+    - joerchan
```
Comment (carlescufi) [inline MAINTAINERS.yml:2302]: Please add @SebastianBoe as well

**Review by carlescufi — DISMISSED** [review]
Thank you!

File: MAINTAINERS.yml (hunk 4 above)
Comment (npal-cy) [reply MAINTAINERS.yml:1908]: it should be @ifyall for both: hal_cypress/hal_infineon

File: MAINTAINERS.yml (hunk 5 above)
Comment (npal-cy) [reply MAINTAINERS.yml:1940]: it should be @ifyall for both: hal_cypress/hal_infineon

File: MAINTAINERS.yml (hunk 7 above)
Comment (brgl) [reply MAINTAINERS.yml:2136]: Sure, would be great. I would also love to be able to add labels to PRs and launch CI workflows for first-time contributors in the main zephyr tree if possible.

File: MAINTAINERS.yml (hunk 3 above)
Comment (nashif) [reply MAINTAINERS.yml:1892]: how was this determined? it is not obsolete.

File: MAINTAINERS.yml [hunk 13]
```diff
@@ -1841,6 +1838,489 @@ West:
...
+    - modules/Kconfig.civetweb
+  labels:
+    - manifest-civetweb
+    - "area: Networking"
+    - "area: civetweb"
+
+"West project: cmsis":
+  status: odd fixes
+  collaborators:
+    - nashif
```
Comment (nashif) [inline MAINTAINERS.yml:1864]: not me

File: MAINTAINERS.yml [hunk 14]
```diff
@@ -1841,6 +1838,489 @@ West:
...
+  status: maintained
+  maintainers:
+    - Mani-Sadhasivam
+  files:
+    - modules/loramac-node/
+  labels:
+    - manifest-loramac-node
+
+"West project: lvgl":
+  status: odd fixes
```
Comment (brgl) [inline MAINTAINERS.yml:2135]: If you want me as maintainer then you can change the status to `maintained`

File: MAINTAINERS.yml (hunk 3 above)
Comment (mbolivar-nordic) [reply MAINTAINERS.yml:1892]: @nashif who is the collaborator then? Does it have any?

File: MAINTAINERS.yml (hunk 14 above)
Comment (stephanosio) [reply MAINTAINERS.yml:2135]: maybe @microbuilder (ARM maintainer) and @povergoing (from ARM)

File: MAINTAINERS.yml (hunk 13 above)
[the remark this answers is not in the document]
Comment (stephanosio) [reply MAINTAINERS.yml:1864]: maybe @microbuilder (ARM maintainer) and @povergoing (from ARM) as collaborators

File: MAINTAINERS.yml (hunk 3 above)
Comment (nashif) [reply MAINTAINERS.yml:1892]: you can put me.

File: MAINTAINERS.yml [hunk 15]
```diff
@@ -1841,6 +1838,489 @@ West:
...
+    - WealianLiao
+  collaborators:
+    - MulinChao
+    - ChiHuaL
+    - WealianLiao
+    - sjg20
+    - keith-zephyr
+    - jackrosenthal
+    - fabiobaltieri
+    - yperess
```
Comment (keith-zephyr) [inline MAINTAINERS.yml:1986]: The hal_nuvoton module is for older EC chips not supported by Chromebooks.  The newer chips are in the tree. So it's probably better to only include the Nuvoton folks and you can remove these individuals
```
    - sjg20
    - keith-zephyr
    - jackrosenthal
    - fabiobaltieri
    - yperess
    ```

File: MAINTAINERS.yml (hunk 10 above)
Comment (microbuilder) [reply MAINTAINERS.yml:2281]: @theotherjimmy as well

File: MAINTAINERS.yml (hunk 11 above)
Comment (microbuilder) [reply MAINTAINERS.yml:2292]: @theotherjimmy as well

File: MAINTAINERS.yml (hunk 12 above)
Comment (microbuilder) [reply MAINTAINERS.yml:2302]: @theotherjimmy as well

File: MAINTAINERS.yml (hunk 13 above)
[the remark this answers is not in the document]
Comment (povergoing) [reply MAINTAINERS.yml:1864]: OK for me

File: MAINTAINERS.yml (hunk 7 above)
Comment (jfischer-no) [reply MAINTAINERS.yml:2136]: Are display collaborators not asked in this regard?

File: MAINTAINERS.yml (hunk 1 above)
Comment (mbolivar-nordic) [reply MAINTAINERS.yml:2261]: > Hmm no, this is not obsolete. There are no viable plans or alternatives to replace it (yet).

Yeah -- I wrote this patch before I wrote https://github.com/zephyrproject-rtos/zephyr/commit/89b07084d668fbc14da5ae1dbc7302b674c5633a. Without that patch, the only thing that fit areas without even a single collaborator was "obsolete". I will update these now that that patch is merged and we are OK with having "odd fixes" be a status for things without collaborators.

File: MAINTAINERS.yml (hunk 3 above)
Comment (mbolivar-nordic) [reply MAINTAINERS.yml:1892]: OK. https://github.com/zephyrproject-rtos/zephyr/pull/44614#discussion_r846299080 for details on how it was determined originally

File: MAINTAINERS.yml (hunk 7 above)
Comment (mbolivar-nordic) [reply MAINTAINERS.yml:2136]: > Are display collaborators not asked in this regard?

@jfischer-no If you are volunteering for the role, please say so clearly and I will add you

**Comment by mbolivar-nordic** [comment]
Thanks for all the reviews. Comments should be addressed, please revisit.

**Comment by mbolivar-nordic** [comment]
Rebased to address merge conflict.

File: MAINTAINERS.yml (hunk 7 above)
Comment (mbolivar-nordic) [reply MAINTAINERS.yml]: > Sure, would be great. I would also love to be able to add labels to PRs and launch CI workflows for first-time contributors in the main zephyr tree if possible.

@brgl awesome! Please read https://docs.zephyrproject.org/latest/project/project_roles.html#contributor and nominate yourself for Triage role as Contributor or as Collaborator as you see fit.

**Review by laurenmurphyx64 — APPROVED** [review]

**Review by MaureenHelm — APPROVED** [review]

**Review by stephanosio — APPROVED** [review]

**Review by keith-zephyr — APPROVED** [review]

**Review by microbuilder — APPROVED** [review]

**Review by nashif — APPROVED** [review]

**Review by povergoing — APPROVED** [review]

**Pull request merged**